doc/tests/manual/reports/230723_heatmap_with_shade.md

```diff
@@ -0,0 +1,238 @@
+  "honors": 0,
+  "visits": 0,
+  "harvested": 0,
+  "privacy": "public",
+  "description": "",
+  "location": null,
+  "owner_id": "361c7c28-020f-4b31-84ea-cc629cc43180",
+  "geometry": {
+    "rings": [
+      [
+        {
+          "x": 0.0,
+          "y": 0.0,
+          "srid": 4326
+        },
+        {
+          "x": 100.0,
+          "y": 0.0,
+          "srid": 4326
+        },
+        {
+          "x": 100.0,
+          "y": 100.0,
+          "srid": 4326
+        },
+        {
+          "x": 50.0,
+          "y": 100.0,
+          "srid": 4326
+        },
+        {
+          "x": 50.0,
+          "y": 50.0,
+          "srid": 4326
+        },
+        {
+          "x": 0.0,
+          "y": 50.0,
+          "srid": 4326
+        },
+        {
+          "x": 0.0,
+          "y": 0.0,
+          "srid": 4326
+        }
+      ]
+    ],
+    "srid": 4326
+  }
+}
+```
+
+5. Create a shading.
+
+```bash
+curl --location 'http://localhost:8080/api/maps/1/layers/shade/shadings' \
+--header 'Content-Type: application/json' \
+--header 'Authorization: Bearer <token>' \
+--data '{
+    "layerId": 3,
+    "shade": "no shade",
+    "geometry": {
+        "rings": [
+            [
+                {
+                    "x": 0.0,
+                    "y": 0.0
+                },
+                {
+                    "x": 20.0,
+                    "y": 0.0
+                },
+                {
+                    "x": 20.0,
+                    "y": 20.0
+                },
+                {
+                    "x": 0.0,
+                    "y": 20.0
+                },
+                {
+                    "x": 0.0,
+                    "y": 0.0
+                }
+            ]
+        ],
+        "srid": 4326
+    },
+    "actionId": "00000000-0000-0000-0000-000000000000"
+}'
+```
+
+The result should be:
+
+```json
+{
+  "id": "21c9ca45-5ff4-492a-a537-7eb64c134613",
+  "layerId": 3,
+  "shade": "no shade",
+  "geometry": {
+    "rings": [
+      [
+        {
+          "x": 0.0,
+          "y": 0.0,
+          "srid": 4326
+        },
+        {
+          "x": 20.0,
+          "y": 0.0,
+          "srid": 4326
+        },
+        {
+          "x": 20.0,
+          "y": 20.0,
+          "srid": 4326
+        },
+        {
+          "x": 0.0,
+          "y": 20.0,
+          "srid": 4326
+        },
+        {
+          "x": 0.0,
+          "y": 0.0,
+          "srid": 4326
+        }
+      ]
+    ],
+    "srid": 4326
+  },
+  "addDate": null,
+  "removeDate": null
+}
+```
+
+6. Execute the request.
+
+```bash
+curl -o file.png --location 'http://localhost:8080/api/maps/1/layers/plants/heatmap?plant_id=1&plant_layer_id=1&shade_layer_id=3' \
+--header 'Authorization: Bearer <token>'
+```
+
+7. Verify:
+
+- The bottom left corner should be transparent, everything else should be green.
+- The top left corner should be green (as there is shade there and plant with id 1 likes shade); the rest should be yellow.
```

Comment:
I don't understand what is unclear about this? This is a manual test, it doesn't have to represent actual data. The only thing that was tested there is if the shade is correctly considered for the plant.
As far as I understand the case study has nothing to do with manual tests.

Reply:
For command 6 I get the error `Plant layer with id 1 not found`. It might be a problem on my side, a script would be easier.

But the main problem I have is that all the scaling in the head is not easy. It is not obvious which coordinates will be shown in the bottom left corner.

> As far as I understand the case study has nothing to do with manual tests.

(Manual) tests are constructed to be minimal and still test all interesting corner cases. A case study is realistic: it might not have all corner cases (would be a big plus, though, if it has) but has many different standard cases. Beside from that, they are more or less the same: they demonstrate that some code works.